### DISCUSSION TITLE
Spring for GraphQL

### DISCUSSION THREAD

**Discussion by anbusampath** [graphql-java-kickstart/graphql-spring-boot#826] [Q&A]
I am new to GraphQL, while looking at some project 
[graphql-spring-boot](https://github.com/graphql-java-kickstart/graphql-spring-boot) is used. Also noticed Spring Team planning to release Spring for GraphQL. How these projects are different from usage's prespective.

**Comment by ZohebMOPO** [comment]
In my opinion graphql-spring-boot is a framework that you can use and you don't need to configure low level stuff and don't need to write stuff from scratch. But if the spring team is gonna release for graphql, it will be a plugin most probably and you need to configure things on your own. Correct me if I am wrong

**Reply by anbusampath** [reply]
After posting the question, I did some more reading both graphql-spring-boot and Spring for GraphQL with Spring Boot starter trying to achieve the same outcome to provide higher level abstraction for users instead of exposing Graphql-java low level apis. Below image from youtube talk - https://www.youtube.com/watch?v=CnHZnt51_m8&


<img width="1269" alt="image" src="https://user-images.githubusercontent.com/1268973/172436557-63f38f12-b008-4272-974c-f01d529ed8b9.png">

**Reply by ZohebMOPO** [reply]
Oh I see. So it's gonna replace this one?

**Reply by anbusampath** [reply]
I would not say replace, both are different implementation to achieve same. It's upto the maintainers to decide. https://github.com/graphql-java/graphql-java-spring - this project archived in favour of Spring for GraphQL.

Another interesting read is that graphqj-java-kickstart project spun out of grapql-java as well. https://www.graphql-java.com/blog/moving-projects

It's good to have more than choice to pick.

**Reply by raihanameen** [reply]
I tried out the first release of Spring for GraphQL, and I was pleasantly surprised by how simple the creators have made it for us developers..

**Reply by anbusampath** [reply]
> I tried out the first release of Spring for GraphQL, and I was pleasantly surprised by how simple the creators have made it for us developers..

Spring/Spring Boot always simplifies the things for developers(Magic).
